grid.cpp

```diff
@@ -211,8 +215,20 @@ void initializeGrids(
       // Each initialization has to be independent to avoid threading problems 
 
       // Allow the project to set up data structures for it's setCell calls
-      project.setupBeforeSetCell(cells);
-      
+      bool needCurl=false;
+      project.setupBeforeSetCell(cells, mpiGrid, needCurl);
+      if (needCurl==true) {
+	 // Communicate the perturbed B-fields read from the start file over to FSgrid
+	 feedPerBIntoFsGrid(mpiGrid, cells, perBGrid);
+	 perBGrid.updateGhostCells();
+	 // Calculate volumetric derivatives of B for curl
+         calculateDerivativesOnlyPerB(perBGrid, dPerBGrid, technicalGrid);
```

Comment:
I think that dPerBGrid might need a ghost update here before going into calculateVolumeAveragedFields.

Reply:
calculateVolumeAveragedFields doesn't seem to be doing it.